pkg/compiler/consts.go

```diff
@@ -297,6 +300,9 @@ func (comp *compiler) patchConst(val *uint64, id *string, consts map[string]uint
 		*val = v
 		return true
 	}
+	if _, isFlag := comp.intFlags[*id]; isFlag {
```

Comment:
It's similar to the `addConst` case you commented: because our type argument has kind `kindInt | kindIdent`, we may end up detecting this identifier has a missing constant. So if we find an identifier that is actually a flags name, we should catch it here to avoid treating it as a constant.

For example, without these three lines, we get error:
```
bin/syz-sysgen
struct bpf_insn_ldst_t[bpf_ldst_insn, bpf_ldst_size, bpf_ldst_mode, flags[bpf_reg, int8:4], flags[bpf_reg, int8:4], flags[bpf_insn_offsets, int16], flags[bpf_insn_immediates, int32]] is unsupported on all arches (typo?)
```
That's because the type template is:
```
type bpf_insn_ldst_t[CLASS, SZ, MODE, DST, SRC, OFF, IMM] {
	code_class	int8:3[CLASS]
	code_size	int8:2[SZ]
	code_mode	int8:3[MODE]
	dst		DST
	src		SRC
	off		OFF
	imm		IMM
}
```
and `int8:3[bpf_ldst_insn]` (among others) causes `bpf_ldst_insn` to be detected as a constant for which we don't have a value.